WalletWasabi/Wallets/FileSystemBlockRepository.cs

```diff
@@ -33,6 +34,8 @@ public FileSystemBlockRepository(string blocksFolderPath, Network network, long
 	public string BlocksFolderPath { get; }
 	private Network Network { get; }
 	private AsyncLock BlockFolderLock { get; } = new AsyncLock();
+	public List<Block> BlockQueue { get; } = new();
+	public object BlockQueueLock { get; } = new();
```

Comment:
We are using this lock in async context all the place. It is generally bad practice to mix async and sync. Replace this with AsyncLock.